src/override/rust.cc

```diff
@@ -9,47 +9,199 @@
 
 using namespace snmalloc;
 
-extern "C" SNMALLOC_EXPORT void*
-  SNMALLOC_NAME_MANGLE(rust_alloc)(size_t alignment, size_t size)
+namespace
 {
-  return ThreadAlloc::get().alloc(aligned_size(alignment, size));
-}
+  enum class SizeChange
```

Comment:
Please can you add a comment explaining (and, ideally, linking to) the Rust APIs that this is intended to expose?